tests/Unit/HasRolesTest.php

```
@@ -0,0 +1,495 @@
+        $this->client->setMockRoles([
+            ['id' => '1', 'key' => 'role:admin', 'name' => 'Role Admin'],
+            ['id' => '2', 'key' => 'role.editor', 'name' => 'Role Editor'],
+            ['id' => '3', 'key' => 'role-viewer', 'name' => 'Role Viewer'],
+        ]);
+
+        $this->assertTrue($this->client->hasRoles(['role:admin', 'role.editor', 'role-viewer']));
+    }
+
+    public function testEarlyExitOnFirstFailure(): void
+    {
+        $callCount = 0;
```

PHPMD (2.15.0)

Avoid unused local variables such as '$callCount'. (undefined)

(UnusedLocalVariable)